src/app/cast.js

```diff
@@ -1,64 +1,62 @@
+const LOG_PREFIX = 'MessageInterceptor:';
+const castPlayer = new CastPlayer();
 
 logger.setLevel(process.env.LOG_LEVEL);
 UIManager.setStatusCast(UIManager.status.LOADING);
-
-// create a new instance of the Media Manager to interact with the sender app request
-mediaManager = new cast.receiver.MediaManager(castPlayer);
-
-// backup of the original onLoad event handler to use it after our custom logic
-mediaManager.origOnLoad = mediaManager.onLoad;
-
-/**
- * onLoad assign a new implementation of the onLoad event that can sets the needed information from the 
- * incoming event
- */
-mediaManager.onLoad = function(e) {
-    logger.info("MediaManager:onLoad", e);
-    castPlayer.setAsset(e.data.media.customData);
-    mediaManager.origOnLoad(e);
-}
-
-
-// backup of the original onPause event handler to use it after our custom logic
-mediaManager.origPause = mediaManager.onPause;
-
-/**
- * onPause overrides the default behaviour to setup the default timeout when an assets it's getting paused 
- */
-mediaManager.onPause = function(e) {
-    logger.info("MediaManager:onPause", e);
+const context = cast.framework.CastReceiverContext.getInstance();
+const playerManager = context.getPlayerManager();
+
+playerManager.setMessageInterceptor(cast.framework.messages.MessageType.LOAD, loadRequestData => {
+    logger.info(LOG_PREFIX, "LOAD", loadRequestData);
+    castPlayer.setAsset(loadRequestData.media.customData);
+    castPlayer.load();
+    playerManager.setMediaElement(castPlayer.OOPlayer);
+    return loadRequestData;
+});
+
+playerManager.setMessageInterceptor(cast.framework.messages.MessageType.PAUSE, requestData => {
+    logger.info(LOG_PREFIX, "PAUSE", requestData);
     Timer.setIdle(Timer.TIMEOUT.PAUSED);
-    mediaManager.origPause(e);
-}
-
-
-// backup of the original onPlay event handler to use it after our custom logic
-mediaManager.origPlay = mediaManager.onPlay;
+    castPlayer.pause();
+    return requestData;
+});
 
-/**
- * onPlay override the original logic for play just to clear the timeout, if one was created 
- */
-mediaManager.onPlay = function(e) {
-    logger.info("MediaManager:onPlay", e);
+playerManager.setMessageInterceptor(cast.framework.messages.MessageType.PLAY, requestData => {
+    logger.info(LOG_PREFIX, "PLAY", requestData);
     Timer.setIdle();
-    mediaManager.origPlay(e);
-}
-
-
-castManager.start();
\ No newline at end of file
+    castPlayer.play();
+    return requestData;
+});
+
+playerManager.setMessageInterceptor(cast.framework.messages.MessageType.SET_VOLUME, volumeRequestData => {
+    logger.info(LOG_PREFIX, "SET_VOLUME", volumeRequestData);
+    castPlayer.setVolume(volumeRequestData.volume);
+    return volumeRequestData;
+});
+
+playerManager.setMessageInterceptor(cast.framework.messages.MessageType.SEEK, seekRequestData => {
+    logger.info(LOG_PREFIX, "SEEK", seekRequestData);
+    castPlayer.seek(seekRequestData.currentTime);
+    castPlayer.onSeeked(_, seekRequestData.currentTime);
```

Comment:
This shouldn't be called here. The player will trigger the "seeked" event after it finished to buffer the asset at the given time.

Reply:
Yes, it's obviously that callback should be fired by message bus. But i have a problem, that this callback doesn't fire, and i have not enough knowledge in web player api to understand why. Maybe u have some idea?